### ISSUE TITLE
User profiles

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#3389]
Currently, tapping a user's name (when seeing a picture) does nothing.
It would be great if that could open an activity that shows:
- This user's level and statistics (reuse the current Achievements activity? Or even transform the Achievements activity to just show one's own profile activity?)
- A "view in browser" button that shows a link to this user's web page, for instance https://commons.wikimedia.org/wiki/User:Syced
- A gallery of this user's pictures, just like in the home activity.

If users are conscious that they have a profile page and that other people can look at it, maybe they will be encouraged to contribute better/more?
If we do it right, this could even give ideas to the website version of Commons.

**Comment by animeshk08** [comment]
I would like to work on this issue. @nicolas-raoul .
I think we can show the user's webpage on clicking the username. It would encourage them to upload more pictures and also get more details of the images they have already uploaded

**Comment by nicolas-raoul** [comment]
@animeshk08 Thanks for getting involved! Are you sure you satisfy this condition, though? :-)

> Issues labeled "enhancement" will usually only be assigned to contributors with >5 previous contributions to this repo.
https://github.com/commons-app/apps-android-commons/wiki/Volunteers-welcome%21#developers

**Comment by misaochan** [comment]
I love this idea! @nicolas-raoul Do you know how it would tie in with the Leaderboard that is due to be implemented in GSoC? It seems to me like some of the logic could be reused.

**Comment by misaochan** [comment]
Also, I wonder if it is possible to pull barnstars from the user's page? If it is, this could tie in nicely with #2428 , so we could award a barnstar to winners, and it would be viewable in their profile page.

**Comment by neslihanturan** [comment]
Hi all, here I my suggested mockups for this feature. 
1- Where to reach a user profile from:
![howToGetThere](https://user-images.githubusercontent.com/3127881/116046139-717ae600-a67b-11eb-9a6f-1d9afe8d7889.png)

2- What will happen when we tap on user name from one of these 3 access points:


Looking forward to hear your feedbacks :)

**Comment by nicolas-raoul** [comment]
Should it be possible to reach a user from the leaderboard? I wonder who is this Vojtech who outranked me ;-)

Will you add a "View user page" in the three-dots menu, similar to "View file page" and "View item page"?

**Comment by neslihanturan** [comment]
Thanks for pointing out important points @nicolas-raoul , I edited the designs accordingly in my previous comment. How did you find the updated versions?

**Comment by nicolas-raoul** [comment]
About the 4th screenshot of section 1: I agree that tapping "Fae" should open Fae's profile.

About the 5th screenshot of section 1: I don't think a search loop widget is necessary here though, as it is already present in Explore.

About the 6th screenshot of section 1: I think that "View user page" would be more relevant in section 2 (achievements/leaderboard/contributions screen).

**Comment by misaochan** [comment]
Hi @neslihanturan , thanks for the mockups! :) 

> 1- Where to reach a user profile from:

I think this is a very ambitious plan and perhaps it may be better to stick to something a bit more basic for the initial prototype. ;)  Increasing the access points increases the potential for issues and the need for debugging, besides adding UI complexity. I would vote for only item 1, 4, and 6, and not implement the rest. Especially with item 2, it is too easy to mis-tap when you want to actually view the image instead.

> 2- What will happen when we tap on user name from one of these 6 access points:

Looks good to me!

**Comment by ashishkumar468** [comment]
To summarise the expected behavior:
Clicks on the author names in places mentioned in Screenshots 1,4 & 6 would open the Profile-Section for that particular user where we will be showing the Achievements, Leaderboard and the ContributionsList of that user. Question: Will we be supporting clicks on contributions?

**Comment by misaochan** [comment]
> Question: Will we be supporting clicks on contributions? @misaochan 

I don't think we need to have this in the initial rollout of this feature. It can be an enhancement for the future. :)
